perly.y

```diff
@@ -913,7 +913,7 @@ sigscalarelem:
                                             "follows optional parameter");
                             }
 
-                            $$ = var ? newSTATEOP(0, NULL, var) : NULL;
+                            $$ = var ? newSTATEOP(OPf_FORCE_NEXTSTATE, NULL, var) : NULL;
```

Comment:
This means that in the debugger it now won't be single-steppable even if there is a defaulting expression. It's hard to predict what users might want here, but maybe it's best to keep the DBSTATE if there is a defop? So maybe

```c
defop ? 0 : OPf_FORCE_NEXTSTATE
```

for the flags?

Reply:
That would get annoying in the same way as in #22602 if you had more than a couple of default valued arguments.

I think the ideal experience would be to make the debugger display each argument at the signature step points as you step through the signature.

Maybe these breakpoints could be optional, but it would need to be controllable at runtime (ie. need a flag on the dbstate), and I expect discoverability of such an option would be bad.

I'm not sure of a good approach for the debugger to get the variable set by the argelem.